File: UI/musicshop_admin/lib/models/abstract/product.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:musicshop_admin/models/Abstract/base_model.dart';

part 'product.g.dart';

@JsonSerializable()
class Product extends BaseModel {
  String? ProductNumber;
  int? ProductImageId;
  int? BrandId;
  String? Model;
  double? Price;
  String? Description;
  DateTime? CreatedAt;
  DateTime? UpdatedAt;

  Product();

  factory Product.fromJson(Map<String, dynamic> json) =>
      _$ProductFromJson(json);

  Map<String, dynamic> toJson() => _$ProductToJson(this);
}

File: MusicShop.Model/BaseModel/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicShop.Model.BaseModel
{
    public class Product : BaseClassModel
    {
        public string? ProductNumber { get; set; }
        public Brand? Brand { get; set; }
        public string? Model { get; set; }
        public decimal? Price { get; set; }
        public string? Description { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public byte[]? ProductImage { get; set; }
        public string? Type { get; set; }
    }
}
